http-server-netty/src/main/java/io/micronaut/http/server/netty/body/SystemFileBodyWriter.java

```diff
@@ -113,34 +124,22 @@ public void writeTo(HttpRequest<?> request, MutableHttpResponse<SystemFile> resp
 
                 // Write the request data
                 final DefaultHttpResponse finalResponse = new DefaultHttpResponse(nettyResponse.getNettyHttpVersion(), nettyResponse.getNettyHttpStatus(), nettyResponse.getNettyHeaders());
-                writeFile(systemFile, nettyContext, position, contentLength, finalResponse);
+
+                File file = systemFile.getFile();
+                InputStream is;
+                try {
+                    is = new FileInputStream(file);
```

Comment:
Who closes this?

Reply:
it is passed on to PipeliningServerHandler where BlockingOutboundHandler.work closes it using try-with-resources